### PR TITLE
Make it possible to point to a different maskfile

### PR DESCRIPTION
**Pull request by felipesere**
This PR adds the ability to specify `--maskfile` or `-m` followed by a path to a maskfile.

I had to work around the fact that clap will still parse all of the arguments from `argv`, so I had to add a subcommand to match `--maskfile` (see the comment below).

Other than that I added a the dev-dependencies to start writing integration tests for the binary.

Let me know what you think!

### CODE REVIEW

**Comment by felipesere** [comment]
Solves #7

**Comment by jakedeichert** [comment]
Hey Felipe, thanks for doing this!

I've got a busy weekend ahead, but i'll try to give this another look tonight or late tomorrow. Looks good at first glance 👍

File: src/main.rs [hunk 1]
```diff
@@ -60,7 +71,17 @@ fn build_subcommands<'a, 'b>(
         }
         cli_app = cli_app.subcommand(subcmd);
     }
-    cli_app
+
+    // This is needed to prevent clap from complaining. It should be removed once
+    // clap 3.x is released. See https://github.com/clap-rs/clap/issues/748
+    let maskfile = Arg::with_name("maskfile")
+        .help("Points to a maskfile to use")
```
Comment (jakedeichert) [inline src/main.rs]: I think specifying this is a "path" might be a little more explicit :)

```suggestion
        .help("Path to a different maskfile you want to use")
```

File: src/main.rs [hunk 2]
```diff
@@ -60,7 +71,17 @@ fn build_subcommands<'a, 'b>(
         }
         cli_app = cli_app.subcommand(subcmd);
     }
-    cli_app
+
+    // This is needed to prevent clap from complaining. It should be removed once
+    // clap 3.x is released. See https://github.com/clap-rs/clap/issues/748
+    let maskfile = Arg::with_name("maskfile")
```
Comment (jakedeichert) [inline src/main.rs]: Just a minor nitpick 👍 
```suggestion
    let custom_maskfile_path = Arg::with_name("maskfile")
```

File: Cargo.toml [hunk 3]
```diff
@@ -18,3 +18,7 @@ path = "src/lib.rs"
 [dependencies]
 clap = "2.33.0"                                                         # https://github.com/clap-rs/clap
 pulldown-cmark = { version = "0.5", default-features = false }          # https://github.com/raphlinus/pulldown-cmark
+
+[dev-dependencies]
+assert_cmd = "0.11.1"
```
Comment (jakedeichert) [inline Cargo.toml]: I haven't used these packages before but they look useful! I have a habit of linking to the github repo though, just for convenience :)

```suggestion
assert_cmd = "0.11.1"                                                   # https://github.com/assert-rs/assert_cmd
```

File: Cargo.toml [hunk 4]
```diff
@@ -18,3 +18,7 @@ path = "src/lib.rs"
...
+assert_fs = "0.11.3"
```
Comment (jakedeichert) [inline Cargo.toml]:
```suggestion
assert_fs = "0.11.3"                                                    # https://github.com/assert-rs/assert_fs
```

File: src/main.rs [hunk 5]
```diff
@@ -7,7 +8,17 @@ use clap::{
...
-    let maskfile = mask::loader::read_maskfile(Path::new("."));
+    let args: Vec<String> = env::args().collect();
+
+    let maybe_maskfile = args.get(1);
+    let maybe_path = args.get(2);
+
+    let maskfile_path = match (maybe_maskfile, maybe_path) {
+        (Some(a), Some(path)) if a == "--maskfile" => Path::new(path),
+        _ => Path::new("./maskfile.md"),
+    };
```
Comment (jakedeichert) [inline src/main.rs]: This looks great 👍

File: tests/integration_test.rs [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+use assert_fs::prelude::*;
+use std::process::Command;
+
+#[test]
+fn specifying_a_maskfile_in_a_different_dir() {
+    let temp = assert_fs::TempDir::new().unwrap();
+    let maskfile = temp.child("maskfile.md");
+
+    maskfile.write_str("
+# Integartion tests
```
Comment (jakedeichert) [inline tests/integration_test.rs]: Minor typo!
```suggestion
# Integration tests
```

File: tests/integration_test.rs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+<!-- a few more details -->
+")
+        .unwrap();
+
+
+    let maskfile_path = maskfile.path().to_str().unwrap();
+
+    let mut mask = Command::cargo_bin(crate_name!()).unwrap();
+    mask.arg("--maskfile").arg(maskfile_path).arg("--help").assert().success();
```
Comment (jakedeichert) [inline tests/integration_test.rs]: Wow. Testing with assert_cmd is so simple. This is great :100:

File: src/main.rs [hunk 8]
```diff
@@ -60,7 +71,17 @@ fn build_subcommands<'a, 'b>(
...
+    // This is needed to prevent clap from complaining. It should be removed once
+    // clap 3.x is released. See https://github.com/clap-rs/clap/issues/748
+    let custom_maskfile_path = Arg::with_name("maskfile")
+        .help("Path to a different maskfile you want to use")
+        .short("m")
+        .long("maskfile")
+        .takes_value(true)
+        .multiple(false);
+
+    cli_app.arg(maskfile)
```
Comment (jakedeichert) [inline src/main.rs]: Whoops i forgot to suggest this.

```suggestion
    cli_app.arg(custom_maskfile_path)
```

File: src/main.rs (hunk 5 above)
Comment (felipesere) [reply src/main.rs]: Thanks :)

File: tests/integration_test.rs (hunk 7 above)
Comment (felipesere) [reply tests/integration_test.rs]: I know! Its almost a luxury now!
In the other branch/PR I've added a bit more of the plumbing to match the domain/wording of mask.

**Comment by felipesere** [comment]
All changes should be up to date now.

File: tests/integration_test.rs (hunk 7 above)
Comment (jakedeichert) [reply tests/integration_test.rs]: Sounds great. I'll be reviewing that shortly 👍

**Comment by jakedeichert** [comment]
Thanks for being so quick! 😄

**Pull request merged**